### DISCUSSION TITLE
`config_hash` in the API endpoints

### DISCUSSION THREAD

**Discussion by anusha2310-netizen** [langchain-ai/langserve#229] [Q&A] [answered]
What is this config_hash
string
(query) parameter in fastapi schema in langserve and what is feedback part in that?

**Comment by eyurtsev** [comment, marked as answer]
Hi @anusha2310-netizen,

The `config_hash` parameter is a compressed json string that contains the config payload. If it's uncompressed and decoded with json, it'll be a config dictionary.

This parameter is not one that you should be setting manual. It's meant to be used in conjunction with the LangServe playground. 

If you create a `configurable` runnable (see for example: https://python.langchain.com/docs/expression_language/how_to/configure) 

The LangServe playground will show a widget with configuration options:

<p align="center">
<img src="https://github.com/langchain-ai/langserve/assets/3205522/09f22850-924f-44ea-8e64-7082da7d5e3b" width="50%"/>
</p>

Once it's configured, you can *share* the configuration using the share button:

<p align="center">
<img src="https://github.com/langchain-ai/langserve/assets/3205522/16d49ce5-6bd1-40b0-8e41-1ecd0e7edab5" width="50%"/>
</p>

The share button will have a value of the `config_hash` that contains the configuration.

The scenario in which you want to use this:

1. You are working in a team setting and have deployed a large complex runnable.
2. One of the people working on the team can use the LangServe playground UI to configure the runnable to work as desired (e.g., edit the prompt)
3. They can then generate a link to the configuration and share it with the team.

The link can be used access the pre-configured runnable from the web-browser. The config hash can also be used with the python or js sdks and be passed as the configuration for the runnable.

**Reply by anusha2310-netizen** [reply]
Thank you @eyurtsev
